### ISSUE TITLE
[HW2] While testing, can we just use the data in InMemoryDataService (instead of mocking responses)?

### ISSUE THREAD

**Issue by lasagnaphil** [swsnu/swppfall2017#44]
Can we just use the data in InMemoryDataService rather than mocking reponses one by one? It seems to me that InMemoryDataService in itself is a mock backend, so there is no need to mock responses by ourselves.
(Example code: https://github.com/angular/in-memory-web-api/blob/master/src/app/hero.service.spec.ts)

I have tried the latter option, but without any success. I have tried using MockBackend to mock responses from http requests (as in the lab ppt). However, even after several tries, the http response I've got from the MockBackend is empty. I am currently suspecting that the problem is because InMemoryWebAPI intercepts those http requests through some kind of dark-magic.

**Comment by wonook** [comment]
Yes, as long as the tests perform what they are intended for, it will be fine to use any kind of mock data.

**Comment by wonook** [comment]
By the way, I think the problem might be that what we have gone through the practice session didn't require us to go in the _data_ object to access data (we only needed to write .json(), but with in-memory data service we need to write json().data). If the mock data for the tests are made in the same way, with a _data_ object, such probem wouldn't occur.

**Comment by SoheeLee** [comment]
Is the difference you are referring to the difference between
```
getHeroes(): Promise<Hero[]> {
    return this.http.get(this.heroesUrl)
               .toPromise()
               .then(response => response.json() as Hero[])
               .catch(this.handleError);
  }
```
and
```
getHeroes(): Promise<Hero[]> {
  return this.http.get(this.heroesUrl)
             .toPromise()
             .then(response => response.json().data as Hero[])
             .catch(this.handleError);
}
```
? How can I make mock data for the test with a data object?

**Comment by wonook** [comment]
@SoheeLee exactly. 
```
      let backend: MockBackend;
      let service: HeroService;
      let fakeHeroes: Hero[];
      let response: Response;

      beforeEach(inject([Http, XHRBackend], (http: Http, be: MockBackend) => {
        backend = be;
        service = new HeroService(http);
        fakeHeroes = makeHeroData();
        let options = new ResponseOptions({status: 200, body: fakeHeroes});
        response = new Response(options);
      }));
```
Notice the code above from our practice session. If you modify it like:
```
        let options = new ResponseOptions({status: 200, body: {data: fakeHeroes}});
```
It would give the same results as our mock backend.

**Comment by lasagnaphil** [comment]
This worked perfectly well for me, thank you! I'm sorry for the late reply....
